app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreRules.tsx

```diff
@@ -155,6 +155,9 @@ const ExploreRules: FC = () => {
     [groups, types, states, searchInput]
   );
 
+  if (allTypes.size === types.length) types.splice(0, types.length);
+  if (allStates.size === states.length) states.splice(0, states.length);
```

Comment:
It would be better to replace `splice` with `setTypes([])` / `setStates([])`, since mutating state directly may prevent React from triggering a re-render.